### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by Mzuha**

### CODE REVIEW

**Review by Basilius-Third — APPROVED** [review]
Максимум компактности и красоты

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 import model.Candidate;
+import model.Cat;
 import model.Person;
 
 public class StreamPractice {
+
+    private static final String NUMBER_SPLIT_REGEX = ",";
+    private CandidateValidator candidateValidator;
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:15]:
```suggestion
    private Predicate<Candidate> candidateValidator;
```

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -14,7 +26,13 @@
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        OptionalInt min = numbers.stream()
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:29]: Feel free to return the result directly without instantiating the new variable.

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -75,6 +111,10 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(c -> candidateValidator.test(c))
```
Comment (Sofasmile) [inline src/main/java/practice/StreamPractice.java:115]: it's enough to pass object of predicate into filter operation

```suggestion
                .filter(candidateValidator)
```

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**